pinot-core/src/main/java/com/linkedin/pinot/core/realtime/impl/dictionary/StringMutableDictionary.java

```diff
@@ -15,42 +15,53 @@
  */
 package com.linkedin.pinot.core.realtime.impl.dictionary;
 
+import it.unimi.dsi.fastutil.ints.Int2ObjectMap;
+import it.unimi.dsi.fastutil.ints.Int2ObjectOpenHashMap;
+import it.unimi.dsi.fastutil.ints.IntSet;
+import it.unimi.dsi.fastutil.objects.Object2IntMap;
+import it.unimi.dsi.fastutil.objects.Object2IntOpenHashMap;
+import java.util.concurrent.atomic.AtomicInteger;
+
+
 public class StringMutableDictionary extends MutableDictionaryReader {
 
-  private String min = null;
-  private String max = null;
+  String2IntBiMap _dictionaryIdBiMap;
+  private String _column;
+  private String _min = null;
+  private String _max = null;
 
   public StringMutableDictionary(String column) {
-    super(column);
+    _column = column;
+    _dictionaryIdBiMap = new String2IntBiMap();
   }
 
   @Override
   public void index(Object rawValue) {
     if (rawValue instanceof Object[]) {
       for (Object o : (Object[]) rawValue) {
-        addToDictionaryBiMap(o.toString());
+        _dictionaryIdBiMap.put(o.toString());
         updateMinMax(o.toString());
       }
       return;
     }
 
-    addToDictionaryBiMap(rawValue.toString());
+    _dictionaryIdBiMap.put(rawValue.toString());
     updateMinMax(rawValue.toString());
   }
 
   private void updateMinMax(String entry) {
```

Comment:
I have tried to keep the existing functionality as much as possible in the PR to keep it clean. If we deduce this is not needed, I'd prefer to keep that cleanup out of this PR.